src/lm.jl

```diff
@@ -156,7 +156,24 @@ function coeftable(mm::LinearModel)
               ["x$i" for i = 1:size(mm.pp.X, 2)], 4)
 end
 
-predict(mm::LinearModel, newx::Matrix) =  newx * coef(mm)
+# Different methods of prediction for the different factorization types
+getR(x::DensePredChol) = UpperTriangular(x.chol.factors)
+getR(x::DensePredQR) = x.qr[:R]
+
+predict((::LinearModel, newx::Matrix)) = newx * coef(mm)
+
+function predict(mm::LinearModel, newx::Matrix, interval_type::Symbol, level = 0.95)
+    prediction = newx * coef(mm)
+    interval_type == :confint || error("only :confint is currently implemented") #:predint will be implemented
+    R = getR(mm.pp)
+
+    tmp = newx * (R\Diagonal(ones(3,3)))
```

Comment:
I've tried a slightly different tack now with a local function, see if you think that would work.